### PR TITLE
chore: universal model compatible

### PR DESCRIPTION
**Pull request by Franklalalala**
1. update neighbor search to new ase API
2. add scale type choice: no_scale, scale_wo_back_grad, scale_w_back_grad
3. add norm_eps for user controllable eps in the norm module
4. add HamilLossAbsMAE for the ham abs mae
5. add adamW optimizer

## Release Notes

* **New Features**
  * Per-pair cutoff support for neighbor list calculations
  * ABACUS CSR format export for Hamiltonian data
  * AdamW optimizer support

* **Enhancements**
  * Added MAE-based Hamiltonian loss function
  * New tunable hyperparameters for layer normalization (norm_eps) and scaling modes (scale_type)

* **Chores**
  * Code cleanup and optimization in seed setup

### CODE REVIEW

File: dptb/nn/deeptb.py [hunk 1]
```diff
@@ -298,9 +299,11 @@ def forward(self, data: AtomicDataDict.Type):
         data = self.embedding(data)
         if hasattr(self, "overlap") and self.method == "sktb":
             data[AtomicDataDict.EDGE_OVERLAP_KEY] = data[AtomicDataDict.EDGE_FEATURES_KEY]
-        
-        data = self.node_prediction_h(data)
-        data = self.edge_prediction_h(data)
+
+        if self.scale_type != 'no_scale':
+            data = self.node_prediction_h(data)
+            data = self.edge_prediction_h(data)
```
[the remark this answers is not in the document]
Comment (Franklalalala) [reply dptb/nn/deeptb.py]: I'll use this, thank you.
